nd4j/nd4j-backends/nd4j-tests/src/test/java/org/nd4j/linalg/factory/ops/NDBaseTest.java

```diff
@@ -0,0 +1,923 @@
+    }
+
+    @Test
+    public void testLt() {
+        NDBase base = new NDBase();
+        INDArray x1 = Nd4j.zeros(DataType.DOUBLE, 3, 3);
+        INDArray x = Nd4j.ones(DataType.DOUBLE, 3, 3);
+        INDArray y = base.lt(x1, x);
+        INDArray y_exp = Nd4j.createFromArray(new boolean[][]{{true, true, true}, {true, true, true}, {true, true, true}});
+        assertEquals(y, y_exp);
+    }
+
+    @Test
+    public void testScalarLte() {
+        //Scalar gt.
+        NDBase base = new NDBase();
+        INDArray x = Nd4j.zeros(DataType.DOUBLE, 3, 3);
+        INDArray y = base.lte(x, 0.1);
+        INDArray y_exp = Nd4j.createFromArray(new boolean[][]{{true, true, true}, {true, true, true}, {true, true, true}});
+        assertEquals(y, y_exp);
+        //TODO: Passes, but I do not like the detailed memory management in the Op constructor.
+    }
+
+    @Test
+    public void testLte() {
+        NDBase base = new NDBase();
+        INDArray x1 = Nd4j.zeros(DataType.DOUBLE, 3, 3);
+        INDArray x = Nd4j.ones(DataType.DOUBLE, 3, 3);
+        INDArray y = base.lte(x1, x);
+        INDArray y_exp = Nd4j.createFromArray(new boolean[][]{{true, true, true}, {true, true, true}, {true, true, true}});
+        assertEquals(y, y_exp);
+    }
+
+    @Test
+    public void testMatchCondition() {
+        // same test as TestMatchTransformOp,
+        NDBase base = new NDBase();
+        INDArray x = Nd4j.create(new double[] {1, 1, 1, 0, 1, 1});
+        INDArray y = base.matchCondition(x, Conditions.epsEquals(0.0));
+        INDArray y_exp = Nd4j.create(new boolean[] {false, false, false, true, false, false});
+        assertEquals(y, y_exp);
+    }
+
+    @Test
+    public void testMatchConditionCount() {
+        NDBase base = new NDBase();
+        INDArray x = Nd4j.create(new double[] {1, 1, 1, 0, 1, 1});
+        INDArray y = base.matchConditionCount(x, Conditions.epsEquals(0.0));
+        assertEquals(y.toString(), "1"); //TODO: There has to be a proper way to do this.
+
+        x = Nd4j.eye(3);
+        y = base.matchConditionCount(x, Conditions.epsEquals(1.0), true, 1);
+        INDArray y_exp = Nd4j.createFromArray(new Long[][]{{1L}, {1L}, {1L}});
+        assertEquals(y, y_exp);
+
+        y = base.matchConditionCount(x, Conditions.epsEquals(1.0), true, 0);
+        y_exp = Nd4j.createFromArray(new Long[][]{{1L, 1L, 1L}});
+        assertEquals(y, y_exp);
+
+        y = base.matchConditionCount(x, Conditions.epsEquals(1.0), false, 1);
+        y_exp = Nd4j.createFromArray(new Long[]{1L, 1L, 1L});
+        assertEquals(y, y_exp);
+
+        y = base.matchConditionCount(x, Conditions.epsEquals(1.0), false, 0);
+        assertEquals(y, y_exp);
+    }
+
+    @Test
+    public void testMax() {
+        NDBase base = new NDBase();
+        INDArray x = Nd4j.eye(3);
+        INDArray y = base.max(x, 0);
+        INDArray  y_exp = Nd4j.createFromArray(new float[]{1.0f, 1.0f, 1.0f});
```

Comment:
createFromArray (1d) versions are all varargs, so `Nd4j.createFromArray(1.0f, 1.0f, 1.0f)` should work